Kin Android/app/src/main/java/kin/unity/KinPlugin.java

```diff
@@ -174,213 +167,249 @@ public void clearAllAccounts( String clientId )
 		}
 	}
 
 
-	public void getStatus( final String accountId )
+	public void getStatus(final String accountId)
 	{
-		new Thread( () -> {
-			try
-			{
-				int status = _accounts.get( accountId ).getStatusSync();
-				unitySendMessage( "GetStatusSucceeded", callbackToJson( String.valueOf( status ), accountId ) );
+		new Thread(() -> {
+			try {
+				int status = _accounts.get(accountId).getStatusSync();
+				unitySendMessage("GetStatusSucceeded", callbackToJson(String.valueOf(status), accountId));
 			}
-			catch( Exception e )
+			catch (Exception e)
 			{
-				unitySendMessage( "GetStatusFailed", exceptionToJson( e, accountId ) );
-				Log.e( TAG, "GetStatus failed", e );
+				unitySendMessage("GetStatusFailed", exceptionToJson(e, accountId));
+				Log.e(TAG, "GetStatus failed", e);
 			}
-		} ).start();
+		}).start();
 	}
 
 
-	public void getBalance( final String accountId )
+	public void getBalance(final String accountId)
 	{
-		new Thread( () -> {
-			try
-			{
-				Balance balance = _accounts.get( accountId ).getBalanceSync();
-				unitySendMessage( "GetBalanceSucceeded", callbackToJson( balance.value().toString(), accountId ) );
+		new Thread(() -> {
+			try {
+				Balance balance = _accounts.get(accountId).getBalanceSync();
+				unitySendMessage("GetBalanceSucceeded", callbackToJson(balance.value().toString(), accountId));
 			}
-			catch( Exception e )
+			catch (Exception e)
 			{
-				unitySendMessage( "GetBalanceFailed", exceptionToJson( e, accountId ) );
-				Log.e( TAG, "GetBalance failed", e );
+				unitySendMessage("GetBalanceFailed", exceptionToJson(e, accountId));
+				Log.e(TAG, "GetBalance failed", e);
 			}
-		} ).start();
+		}).start();
 	}
 
 
-	public void getMinimumFee( final String clientId )
+	public void getMinimumFee(final String clientId)
 	{
-		new Thread( () -> {
-			try
-			{
-				long fee = _clients.get( clientId ).getMinimumFeeSync();
-				unitySendMessage( "GetMinimumFeeSucceeded", callbackToJson( Long.toString( fee ), clientId ) );
+		new Thread(() -> {
+			try {
+				long fee = _clients.get(clientId).getMinimumFeeSync();
+				unitySendMessage("GetMinimumFeeSucceeded", callbackToJson(Long.toString(fee), clientId));
 			}
-			catch( Exception e )
+			catch (Exception e)
 			{
-				unitySendMessage( "GetMinimumFeeFailed", exceptionToJson( e, clientId ) );
-				Log.e( TAG, "GetMinimumFee failed", e );
+				unitySendMessage("GetMinimumFeeFailed", exceptionToJson(e, clientId));
+				Log.e(TAG, "GetMinimumFee failed", e);
 			}
-		} ).start();
+		}).start();
 	}
 
 
-	public void buildTransaction( final String accountId, final String toAddress, final String kinAmount, final int fee, final String memo )
+	public void buildTransaction(final String accountId, final String toAddress, final String kinAmount, final int fee, final String memo)
 	{
-		new Thread( () -> {
-			try
-			{
-				Log.i( TAG, "Preparing to build transaction. toAddress: " + toAddress + ", memo: " + memo );
+		new Thread(() -> {
```

Comment:
why creating a thread for each operation instead of using async operations of the sdk?

Reply:
Not sure, this is part of the original code that @prime31 wrote